Add WINDOW_RESOLUTION_CHANGE

ZackaryH8 opened the pull request:
I have no idea if I did this right, hopefully I did. If there is any issues, feel free to let me know

ZackaryH8 committed:
Add WINDOW_RESOLUTION_CHANGE

drakeee commented on src/Events/Server/MiscEvents.cpp:
```diff
@@ -167,4 +167,19 @@ REGISTER_LOCAL_EVENT(
+		const alt::CWindowResolutionChangeEvent* event = static_cast<const alt::CWindowResolutionChangeEvent*>(ev);
+		lua_State* L = resourceImpl->GetLuaState();
+
+		lua_pushbaseobject(L, event->GetOldResolution());
```
although it should work with `lua_pushbaseobject` you should use `lua_pushvector` instead because the difference between the two is that pushing a baseobject (`lua_pushbaseobject`) will reference the object in the lua registry so it won't be garbage collected, but with `lua_pushvector` it will not save reference in the registry so it can be garbage collected

drakeee commented on src/Events/Server/MiscEvents.cpp:
```diff
@@ -167,4 +167,19 @@ REGISTER_LOCAL_EVENT(
+		lua_pushbaseobject(L, event->GetOldResolution());
+		lua_pushbaseobject(L, event->GetNewResolution());
+
+		return 1;
```
should be `return 2;` because you are returning two objects from the stack

drakeee commented:
also it is a client event and not a server event, so it should be here instead: https://github.com/drakeee/altv-lua-module/blob/dev/src/Events/Client/MiscEvents.cpp

ZackaryH8 committed:
Move Event and Change to pushvector2

Moved event from server to client and use pushvector2 instead of lua_pushbaseobject

ZackaryH8 committed:
Update SDK

drakeee commented on src/Events/Client/MiscEvents.cpp:
```diff
@@ -77,4 +78,19 @@ REGISTER_LOCAL_EVENT(
+		lua_pushvector2(L, event->GetOldResolution());
+		lua_pushvector2(L, event->GetNewResolution());
+
+		return 1;
```
still `return 1` instead of `return 2`

ZackaryH8 committed:
Return 2 instead of 1
